sklearn/tree/_splitter.pyx

```diff
@@ -1047,11 +1160,10 @@ cdef class BestSparseSplitter(BaseSparseSplitter):
             current.feature = features[f_j]
             self.extract_nnz(current.feature, &end_negative, &start_positive,
                              &is_samples_sorted)
-
             # Sort the positive and negative parts of `Xf`
-            simultaneous_sort(&Xf[start], &samples[start], end_negative - start)
-            if start_positive < end:
-                simultaneous_sort(&Xf[start_positive], &samples[start_positive], end - start_positive)
+            sort(&Xf[start], &samples[start], end_negative - start)
+            sort(&Xf[start_positive], &samples[start_positive],
```

Comment:
The failure in the CI was a bug fixed by @thomasjpfan  here when `start_positive == end`. I think that we should then do:

```suggestion
            if start_positive < end:
                sort(&Xf[start_positive], &samples[start_positive],
```